packages/gatsby/src/utils/webpack.config.js

```diff
@@ -466,6 +469,81 @@ module.exports = async (
   if (stage === `build-javascript`) {
     const componentsCount = store.getState().components.size
 
+    const splitChunks = {
+      chunks: `all`,
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        framework: {
+          chunks: `all`,
+          name: `framework`,
+          // This regex ignores nested copies of framework libraries so they're bundled with their issuer.
+          test: new RegExp(
+            `(?<!node_modules.*)[\\\\/]node_modules[\\\\/](${FRAMEWORK_BUNDLES.join(
+              `|`
+            )})[\\\\/]`
+          ),
+          priority: 40,
+          // Don't let webpack eliminate this chunk (prevents this chunk from becoming a part of the commons chunk)
+          enforce: true,
+        },
+        // if a module is bigger than 160kb from node_modules we make a separate chunk for it
+        lib: {
+          test(module) {
+            return (
+              module.size() > 160000 &&
+              /node_modules[/\\]/.test(module.identifier())
+            )
+          },
+          name(module) {
+            const hash = crypto.createHash(`sha1`)
+            if (module.type === `css/mini-extract`) {
```

Comment:
worth a comment why `css/mini-extract` have different hash updates? also do modules of types others than that specific one could need that as well?